### PR TITLE
ci(docs): linting and formatting

### LINKED ISSUE #1400: Add linter to docs project
Quite often we are seeing pr with whitespace fixes and other formatting issues fixed later example: https://github.com/Bearer/bearer/pull/1399/files

Would be nice to have something in place to enforce good fomatting and consistancy

### PR DESCRIPTION
**Pull request by gotbadger**
## Description
- Add linting and formatting to docs 
- Stop full ci running on doc changes 

closes #1400

## Checklist

- [ ] I've added test coverage that shows my fix or feature works as expected.
- [ ] I've updated or added documentation if required.
- [ ] I've included usage information in the description if CLI behavior was updated or added.
- [ ] PR title follows [Conventional Commits](https://www.conventionalcommits.org/) format

### CODE REVIEW

File: docs/contributing/code.md [hunk 1]
```diff
@@ -32,17 +32,17 @@ go mod download
 
 Finally, we use [direnv](https://direnv.net/) to manage env vars in development. This is primarily used for running tests. You can either:
 
-- Set up your own by using `.envrc.example` as a base. You're unlikely to need to set anything other than `GITHUB_WORKSPACE` for integration tests, so feel free to skip this step or export it manually using: ```export GITHUB_WORKSPACE=`pwd` ```.
+- Set up your own by using `.envrc.example` as a base. You're unlikely to need to set anything other than `GITHUB_WORKSPACE` for integration tests, so feel free to skip this step or export it manually using: ``export GITHUB_WORKSPACE=`pwd` ``.
```
Comment (elsapet) [inline docs/contributing/code.md:35]: Is this correct? From three to two (not one) back ticks?

File: docs/contributing/recipes.md [hunk 2]
```diff
@@ -11,16 +11,16 @@ Recipes are located at `bearer/internal/classification/db/recipes/`.
 ```md
 .
 ├ internal/
-│  └ classification/
-│       └  db/
-│           └  recipes/
+│ └ classification/
+│ └ db/
+│ └ recipes/
```
Comment (elsapet) [inline docs/contributing/recipes.md]: We'll want to keep the original whitespace here I think

File: docs/guides/dataflow.md [hunk 3]
```diff
@@ -119,7 +119,7 @@ Next, the `components` section contains third party services and local or extern
           "line_number": 36
         }
       ]
-    },
+    }
     // ...
```
Comment (elsapet) [inline docs/guides/dataflow.md:123]: Maybe we want the commas to stay here? Not a big one though, I reckon.

**Review by elsapet — APPROVED** [review]
Flagged a few things for confirmation, but overall looks good 👍

File: docs/.eleventy.js [hunk 4]
```diff
@@ -1,96 +1,96 @@
...
-const markdownItAnchor = require("markdown-it-anchor")
-const pluginTOC = require("eleventy-plugin-toc")
-const now = String(Date.now())
-const path = require("path")
-const fs = require("fs")
-const mermaid = require("./_src/_plugins/mermaid")
-const nav = require("./_data/nav")
+} = require("@11ty/eleventy");
+
+const yaml = require("js-yaml");
```
Comment (cfabianski) [inline docs/.eleventy.js]: Could we do the other way around please? I'd rather not have to write the `;` if possible

**Review by cfabianski — CHANGES_REQUESTED** [review]

**Review by cfabianski — APPROVED** [review]

File: docs/guides/dataflow.md (hunk 3 above)
Comment (gotbadger) [reply docs/guides/dataflow.md:123]: Trailing comma is invalid json so think its done right here.

File: .github/workflows/e2e_test.yml [hunk 5]
```diff
@@ -8,6 +8,8 @@ on:
       - opened
       - synchronize
       - reopened
+    paths-ignore:
+      - 'docs/**'
```
Comment (cfabianski) [inline .github/workflows/e2e_test.yml:12]: I have thought about doing this this week! Thanks

File: docs/contributing/code.md (hunk 1 above)
Comment (gotbadger) [reply docs/contributing/code.md:35]: seems good til you can do `` if you need to escape ` :)

Comment (elsapet) [reply docs/contributing/code.md:35]: Niiiice 😎

**Pull request merged**